core/services/run_manager.go

```diff
@@ -178,6 +178,42 @@ func (rm *runManager) CreateErrored(
 	return &run, rm.orm.CreateJobRun(&run)
 }
 
+// If we have seen the same runRequest already, then double the required incoming confs.
+// This is a
+func (rm *runManager) MaybeDoubleMinIncomingConfs(run models.JobRun) (*models.JobRun, error) {
+	if run.RunRequest.RequestID == nil {
+		return &run, logger.NewErrorw("RunManager: expected non-nil run request ID")
+	}
+	if len(run.TaskRuns) == 0 {
+		return &run, logger.NewErrorw("RunManager: expected non-empty task runs")
+	}
+	// We want the maximum number of random task minimum_confirmations
+	// of all job runs with the same run_request.request_id
+	var maxConfs uint32
+	if err := rm.orm.DB.Raw(`
+SELECT coalesce(max(task_runs.minimum_confirmations), 0) FROM job_runs 
+	JOIN run_requests ON job_runs.run_request_id = run_requests.id 
+	JOIN task_runs ON job_runs.id = task_runs.job_run_id
+	JOIN task_specs ON task_runs.task_spec_id = task_specs.id
+	WHERE run_requests.request_id = ? AND task_specs.type = ?
+`, run.RunRequest.RequestID, adapters.TaskTypeRandom).Scan(&maxConfs).Error; err != nil {
+		return &run, err
+	}
+	if maxConfs != 0 {
+		for i := range run.TaskRuns {
+			if run.TaskRuns[i].TaskSpec.Type == adapters.TaskTypeRandom && run.TaskRuns[i].MinRequiredIncomingConfirmations.Valid {
+				logger.Debugw("RunManager: duplicate VRF requestID seen, doubling incoming confirmations",
```

Comment:
maybe worth printing at level Info or even Warn?

Reply:
Also you could print the value of maxConfs

Reply:
yeah I'll make it a warn